### PR TITLE
Convert DownloadTest.java into kotlin

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes: #3062 
Parent Issue: #1826 

Converted DownloadTest.java into kotlin

### CODE REVIEW

**Comment by kelson42** [comment]
. 

 .,,?,....?..

..
,
,.

! .  @MohitMaliFtechiz ready to review?

**Comment by MohitMaliFtechiz** [comment]
> . 
> 
>  .,,?,....?..
> 
> ..
> ,
> ,.
> 
> ! .  @MohitMaliFtechiz ready to review?

Yes.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    } catch (e: Exception) {
+      Assert.fail(
+        """
+  Couldn't find downloaded file 'ray_charles'
+  
+  Original Exception:
+  ${e.localizedMessage}
+  
+  
+        """.trimIndent()
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt:92]: [nit] Remove unnecessary lines.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+/*
+Commented out the following as it uses another Activity.
+TODO Once we find a good way to run cross-activity re-implement
+this functionality in the tests.
+
+    onData(withContent("ray_charles")).inAdapterView(withId(R.id.zimfilelist)).perform(click());
+    openContextualActionModeOverflowMenu();
+    onView(withText("Get Content")).perform(click());
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt:111]: [nit] The documentation looks odd in the idea. It should start with the spaces or horizontally aligned with the function's name.

**Review by gouri-panda — DISMISSED** [review]

File: app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+  @Rule
+  var writePermissionRule: GrantPermissionRule =
+    GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+
+  @Before fun setUp() {
+    // default
+  }
+
+  @Test @Ignore("Broken in 2.5") // TODO: Fix in 3.0
```
Comment (gouri-panda) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt:69]: By the way, We should fix this now. I don't know if the ticket has opened or not.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt (hunk 1 above)
Comment (MohitMaliFtechiz) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt]: Done

File: app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt (hunk 2 above)
Comment (MohitMaliFtechiz) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt]: Done

**Comment by kelson42** [comment]
@gouri-panda This is again ready to review and I just have rebased.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt (hunk 3 above)
Comment (kelson42) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt:69]: I think we have no ticket open. Could you please open one... if we still support 2.5. Do we?

Comment (gouri-panda) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt:69]: Should Fix in https://github.com/kiwix/kiwix-android/issues/3072

**Review by gouri-panda — APPROVED** [review]
@MohitMaliFtechiz I have added two additional commits. Waiting for your review.

**Comment by MohitMaliFtechiz** [comment]
> 

Hi @gouri-panda ,

it's approved by my side.

**Pull request merged**